src/components/CustomInput/CustomInput.tsx

```diff
@@ -1,76 +1,76 @@
-import { forwardRef, useImperativeHandle, useState } from 'react';
+import {
+  ChangeEvent,
+  FocusEvent,
+  forwardRef,
+  ReactNode,
+  useImperativeHandle,
+  useState,
+} from 'react';
 
 import classNames from 'classnames';
 
-import scss from './InputSearch.module.scss';
+import { Icon } from '~components/Icon/Icon';
 
-import alertIcon from './assets/alertIcon.svg';
-import leftIcon from './assets/leftIcon.svg';
-import rightIcon from './assets/rightIcon.svg';
+import scss from './CustomInput.module.scss';
 
-import { TInputProps, TInputComponentRef } from './InputSearch.types';
+import { TInputComponentRef, TInputProps } from './CustomInput.types';
 
-function InputSearch(props: TInputProps, ref: TInputComponentRef) {
+function CustomInput(
+  { error = false, ...props }: TInputProps,
+  ref: TInputComponentRef
+): ReactNode {
   const [value, setValue] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
 
   const labelClass = [scss.label];
   const inputClass = [scss.input];
   const legendClass = [scss.legend];
   const fieldsetClass = [scss.fieldset];
   const containerClass = [scss.container];
-  const [isFocused, setIsFocused] = useState(false);
 
-  const IconAlert = () => {
-    return <img className={scss.icon} src={alertIcon} />;
-  };
-
-  const IconRight = () => {
-    return <img src={rightIcon} />;
+  const handleClear = (): void => {
+    props.onChange?.('');
+    setValue('');
   };
 
-  const handleIcons = () => (
+  const handleIcons = (): ReactNode => (
     <div
       className={scss.iconRight}
       data-testid="clear-button"
       onClick={handleClear}
     >
```

Comment:
se é clicavel precisa ser um `button` pela questão de acessibilidade
```suggestion
    <button
      className={scss.iconRight}
      data-testid="clear-button"
      onClick={handleClear}
    >
```